x-pack/plugins/ml/public/application/explorer/anomalies_map.tsx

```diff
@@ -0,0 +1,246 @@
+
+const MAX_ENTITY_VALUES = 3;
+const COMMON_EMS_LAYER_IDS = [
+  'world_countries',
+  'administrative_regions_lvl2',
+  'usa_zip_codes',
+  'usa_states',
+];
+
+function getAnomalyRows(anomalies: AnomaliesTableRecord[], jobId: string) {
+  const anomalyRows: Record<
+    string,
+    { count: number; entityValue: string; max_severity: number }
+  > = {};
+  for (let i = 0; i < anomalies.length; i++) {
+    const anomaly = anomalies[i];
+    const location = anomaly.entityValue;
+    if (anomaly.jobId !== jobId) continue;
+
+    if (anomalyRows[location] === undefined) {
+      // add it to the object and set count to 1
+      anomalyRows[location] = {
+        count: 1,
+        entityValue: location,
+        max_severity: Math.floor(anomaly.severity),
+      };
+    } else {
+      anomalyRows[location].count += 1;
+      if (anomaly.severity > anomalyRows[location].max_severity) {
+        anomalyRows[location].max_severity = Math.floor(anomaly.severity);
+      }
+    }
+  }
+  return Object.values(anomalyRows);
+}
+
+export const getChoroplethAnomaliesLayer = (
+  anomalies: AnomaliesTableRecord[],
+  { layerId, field, jobId }: MLEMSTermJoinConfig,
+  visible: boolean
+): VectorLayerDescriptor => {
+  return {
+    id: htmlIdGenerator()(),
+    label: `Anomalies count: ${jobId}`,
+    joins: [
+      {
+        // Left join is the id from the type of field (e.g. world_countries)
+        leftField: field,
+        right: {
+          id: 'anomaly_count',
+          type: SOURCE_TYPES.TABLE_SOURCE,
+          __rows: getAnomalyRows(anomalies, jobId),
+          __columns: [
+            {
+              name: 'entityValue',
+              type: 'string',
+            },
+            {
+              name: 'count',
+              type: 'number',
+            },
+            {
+              name: 'max_severity',
+              type: 'number',
+            },
+          ],
+          // Right join/term is the field in the doc you’re trying to join it to (foreign key - e.g. US)
+          term: 'entityValue',
+        },
+      },
+    ],
+    sourceDescriptor: {
+      type: 'EMS_FILE',
+      id: layerId,
+    },
+    style: {
+      type: 'VECTOR',
+      // @ts-ignore missing style properties. Remove once 'VectorLayerDescriptor' type is updated
+      properties: {
+        icon: { type: STYLE_TYPE.STATIC, options: { value: 'marker' } },
+        fillColor: {
+          type: STYLE_TYPE.DYNAMIC,
+          options: {
+            color: 'Blue to Red',
+            colorCategory: 'palette_0',
+            fieldMetaOptions: { isEnabled: true, sigma: 3 },
+            type: COLOR_MAP_TYPE.ORDINAL,
+            field: {
+              name: 'count',
+              origin: FIELD_ORIGIN.JOIN,
+            },
+            useCustomColorRamp: false,
+          },
+        },
+        lineColor: {
+          type: STYLE_TYPE.DYNAMIC,
+          options: { color: '#3d3d3d', fieldMetaOptions: { isEnabled: true } },
```

Comment:
should we import this color from the EUI? and does it depend on the light/dark theme?

Reply:
Updated in d92e417a9fe57db1172fb65ad003fb6742be1ff6